lib/dal/src/change_set.rs

```diff
@@ -320,15 +370,53 @@ impl ChangeSet {
         ctx: &DalContext,
         status: ChangeSetStatus,
     ) -> ChangeSetResult<()> {
+        ctx.txns()
+            .await?
+            .pg()
+            .query_none(
+                "UPDATE change_set_pointers SET status = $2, updated_at = CLOCK_TIMESTAMP() WHERE id = $1",
+                &[&self.id, &status.to_string()],
+            )
+            .await?;
+
+        self.status = status;
         billing_publish::for_change_set_status_update(ctx, self)
             .await
             .map_err(Box::new)?;
+        Ok(())
+    }
 
+    pub async fn request_change_set_approval(&mut self, ctx: &DalContext) -> ChangeSetResult<()> {
+        let user_pk = Self::extract_userid_from_context_or_error(ctx).await?;
+        let status = ChangeSetStatus::NeedsApproval;
         ctx.txns()
             .await?
             .pg()
             .query_none(
-                "UPDATE change_set_pointers SET status = $2, updated_at = CLOCK_TIMESTAMP() WHERE id = $1",
+                "UPDATE change_set_pointers SET merge_requested_by_user_id = $2, merge_requested_at = CLOCK_TIMESTAMP(), status = $3, updated_at = CLOCK_TIMESTAMP() WHERE id = $1",
+                &[&self.id, &user_pk, &status.to_string()],
+            )
+            .await?;
+
+        self.status = status;
+
+        Ok(())
+    }
+
+    /// Set the status to Open, and clear any reviewed/merge requested info
+    pub async fn reopen_change_set(&mut self, ctx: &DalContext) -> ChangeSetResult<()> {
+        let status = ChangeSetStatus::Open;
+        ctx.txns()
+            .await?
+            .pg()
+            .query_none(
+                "UPDATE change_set_pointers
+                SET reviewed_by_user_id = NULL,
+                reviewed_at = NULL,
+                merge_requested_by_user_id = NULL,
+                merge_requested_at = NULL,
```

Comment:
Yep, seems like a good idea to clear these fields out so we don't have prior state on the next approval attempt.